addon-test-support/wait.js

```diff
@@ -1,98 +1,7 @@
-/* globals self */
-
-import { run } from '@ember/runloop';
-
-import { Promise as EmberPromise } from 'rsvp';
-import jQuery from 'jquery';
-
-import Ember from 'ember';
-
-var requests;
-function incrementAjaxPendingRequests(_, xhr) {
-  requests.push(xhr);
-}
-
-function decrementAjaxPendingRequests(_, xhr) {
-  // In most Ember versions to date (current version is 2.16) RSVP promises are
-  // configured to flush in the actions queue of the Ember run loop, however it
-  // is possible that in the future this changes to use "true" micro-task
-  // queues.
-  //
-  // The entire point here, is that _whenever_ promises are resolved, this
-  // counter will decrement. In the specific case of AJAX, this means that any
-  // promises chained off of `$.ajax` will properly have their `.then` called
-  // _before_ this is decremented (and testing continues)
-  EmberPromise.resolve().then(() => {
-    for (var i = 0; i < requests.length; i++) {
-      if (xhr === requests[i]) {
-        requests.splice(i, 1);
-      }
-    }
-  });
-}
-
-export function _teardownAJAXHooks() {
-  if (!jQuery) {
-    return;
-  }
-
-  jQuery(document).off('ajaxSend', incrementAjaxPendingRequests);
-  jQuery(document).off('ajaxComplete', decrementAjaxPendingRequests);
-}
-
-export function _setupAJAXHooks() {
-  requests = [];
-
-  if (!jQuery) {
-    return;
-  }
-
-  jQuery(document).on('ajaxSend', incrementAjaxPendingRequests);
-  jQuery(document).on('ajaxComplete', decrementAjaxPendingRequests);
-}
-
-var _internalCheckWaiters;
-if (Ember.__loader.registry['ember-testing/test/waiters']) {
-  _internalCheckWaiters = Ember.__loader.require('ember-testing/test/waiters').checkWaiters;
-}
-
-function checkWaiters() {
-  if (_internalCheckWaiters) {
-    return _internalCheckWaiters();
-  } else if (Ember.Test.waiters) {
-    if (Ember.Test.waiters.any(([context, callback]) => !callback.call(context))) {
-      return true;
-    }
-  }
-
-  return false;
-}
-
-export default function wait(_options) {
-  var options = _options || {};
-  var waitForTimers = options.hasOwnProperty('waitForTimers') ? options.waitForTimers : true;
-  var waitForAJAX = options.hasOwnProperty('waitForAJAX') ? options.waitForAJAX : true;
-  var waitForWaiters = options.hasOwnProperty('waitForWaiters') ? options.waitForWaiters : true;
-
-  return new EmberPromise(function(resolve) {
-    var watcher = self.setInterval(function() {
-      if (waitForTimers && (run.hasScheduledTimers() || run.currentRunLoop)) {
-        return;
-      }
-
-      if (waitForAJAX && requests && requests.length > 0) {
-        return;
-      }
-
-      if (waitForWaiters && checkWaiters()) {
-        return;
-      }
-
-      // Stop polling
-      self.clearInterval(watcher);
-
-      // Synchronously resolve the promise
-      run(null, resolve);
-    }, 10);
-  });
-}
+export {
+  default,
+  _setupAJAXHooks,
+  _setupPromiseListeners,
+  _teardownAJAXHooks,
+  _teardownPromiseListeners,
+} from './settled';
```

Comment:
this is fine for now, I'm just wondering if we should deprecate importing this module at some point?

Reply:
Ya, totally agreed. Basically, I want to introduce all the new functionality _without_ deprecating anything at first. Then as we get more sure of the new system (e.g. after a little while of using the new API’s by default) we add in all the deprecations. 

I want to be very careful about deprecation spew leading in to Ember 3.0...